pydantic/_internal/_dataclasses.py

```diff
@@ -143,3 +173,41 @@ class B(A):
 
 def is_pydantic_dataclass(_cls: type[Any]) -> TypeGuard[type[PydanticDataclass]]:
     return dataclasses.is_dataclass(_cls) and hasattr(_cls, '__pydantic_validator__')
+
+
+def rebuild_dataclass(
+    cls: type[PydanticDataclass],
+    *,
+    force: bool = False,
+    raise_errors: bool = True,
+    _parent_namespace_depth: int = 2,
+    _types_namespace: dict[str, Any] | None = None,
+) -> bool | None:
+    """
+    Compare with BaseModel.model_rebuild
+
+    TODO: Update this docstring
```

Comment:
Need to update this, but wanted to get some review before doing so as I suspect this may be affected by some requested refactors..